src/Maps/MapWriter.cpp

```diff
@@ -101,16 +90,19 @@ namespace MapWriter {
 		}
 
 		// Outpost 2 map files represent container sizes as 4 byte values
-		void WriteContainerSize(StreamWriter& streamWriter, uint32_t size)
+		void WriteContainerSize(StreamWriter& streamWriter, std::size_t size)
 		{
+			if (size > UINT32_MAX) {
+				throw std::runtime_error("Container size is too large for writing into an Outpost 2 maps.");
+			}
+
 			streamWriter.Write(size);
```

Comment:
This line has been impacted by the function signature update. This write needs to be updated to include a cast. As it stands, it can write a different amount of data between 32-bit and 64-bit builds.

Reply:
Hmm, okay, I'll update soon.